### PR TITLE
Add core package with TS declarations

### PR DESCRIPTION
**Pull request by emmenko**
I've got it to a point where the types compile and the autocompletion is helpful.

image

image

There is still some cleanup to do, but the tests work (the same as we have internally) and this is a good sign :)

### CODE REVIEW

File: core/src/generator.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import type { Overrides } from '@jackfranklin/test-data-bot';
+import type { TGeneratorOptions, TGeneratorResult } from './types';
+
+import { build } from '@jackfranklin/test-data-bot';
+
+// /* TYPES DECLARATIONS FROM @jackfranklin/test-data-bot */
```
Comment (emmenko) [inline core/src/generator.ts:6]: I opened a PR to export the types: https://github.com/jackfranklin/test-data-bot/pull/344

**Comment by emmenko** [comment]
@tdeekens @adnasa this is ready for review, whenever you have time.

PS: if you need a higher level introduction about the implementation and types, let me know and we can schedule a meeting. Thanks

**Comment by tdeekens** [comment]
One high level note. This is valid in regards to using and defining test-data.

```js
const preset = customer.random().id('foo').name('Tobi');

const nicola = preset.name('Nicola');
```

This implies that a prop of a builder can be set twice or overwritten. This should be allowed by the types too. How I see it we have put effort in not allowing this?

**Comment by emmenko** [comment]
I see. So you can potentially do `Builder().name('1').name('2').name('3')`?

**Review by adnasa — APPROVED** [review]
I think this looks good. I like Tobi's suggestion, but I don't know if we want to do this on this PR or on a separate step.

File: core/src/builder.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                  transformed = (transformers?.default?.transform(built) ??
+                    built) as Model;
+                  break;
+                }
+                case 'buildGraphql': {
+                  transformed = (transformers?.graphql?.transform(built) ??
+                    built) as Model;
+                  break;
+                }
+                case 'buildRest': {
```
Comment (tdeekens) [inline core/src/builder.ts:134]: Probably for the best.

**Review by tdeekens — COMMENTED** [review]
Very nice. Nothing to add!

**Review by tdeekens — APPROVED** [review]

**Comment by adnasa** [comment]
😍

**Pull request merged**